src/mainview.tsx

```diff
@@ -382,20 +357,39 @@ export class MainView extends React.Component<IProps, IStates> {
   private _onPointerMove(e: MouseEvent) {
     const rect = this._renderer.domElement.getBoundingClientRect();
 
-    this._pointer.x = e.clientX - rect.left;
-    this._pointer.y = e.clientY - rect.top;
+    this._pointer.x = ((e.clientX - rect.left) / rect.width) * 2 - 1;
+    this._pointer.y = -((e.clientY - rect.top) / rect.height) * 2 + 1;
```

Comment:
I'm just curious, what does this formula mean?

Reply:
It scales the DOM coordinates into viewport coordinates (in the range `[-1, 1]`)

Reply:
Thanks!